xarray/tests/test_groupby.py

```diff
@@ -1813,6 +1813,30 @@ def resample_as_pandas(array, *args, **kwargs):
         with pytest.raises(ValueError):
             reverse.resample(time="1D").mean()
 
+    @pytest.mark.parametrize("use_cftime", [True, False])
+    def test_resample_dtype(self, use_cftime: bool) -> None:
+        if use_cftime and not has_cftime:
+            pytest.skip()
+        array = DataArray(
+            np.arange(10),
+            [
+                (
+                    "time",
+                    xr.date_range(
+                        "2000-01-01", freq="6h", periods=10, use_cftime=use_cftime
+                    ),
+                )
+            ],
+        )
+        test_resample_freqs = (
+            "10min",
+            pd.Timedelta(hours=2),
+            pd.offsets.MonthBegin(),
+            datetime.timedelta(days=1, hours=6),
+        )
+        for freq in test_resample_freqs:
+            array.resample(time=freq)
```

Comment:
Thanks! this is looking quite good.

One request: can we compare the result here to an expected value, possibly generated using pandas directly?

Reply:
Sure, so run a `.mean` on the result here?

Reply:
Sure.